src/dotnet/AzureAI.Proxy/ReverseProxy/ThrottlingHealthPolicy.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Yarp.ReverseProxy.Health;
+using Yarp.ReverseProxy.Model;
+
+namespace AzureAI.Proxy.ReverseProxy;
+
+public class ThrottlingHealthPolicy : IPassiveHealthCheckPolicy
+{
+    public static string ThrottlingPolicyName = "ThrottlingPolicy";
+    private readonly IDestinationHealthUpdater _healthUpdater;
+
+    public ThrottlingHealthPolicy(IDestinationHealthUpdater healthUpdater)
+    {
+        _healthUpdater = healthUpdater;
+    }
+
+    public string Name => ThrottlingPolicyName;
+
+    public void RequestProxied(HttpContext context, ClusterState cluster, DestinationState destination)
+    {
+        var headers = context.Response.Headers;
+
+        if (context.Response.StatusCode is 429 or >= 500)
+        {
+            var retryAfterSeconds = 10;
+
+            if (headers.TryGetValue("Retry-After", out var retryAfterHeader) && retryAfterHeader.Count > 0 && int.TryParse(retryAfterHeader[0], out var retryAfter))
```

Comment:
afaik the retry header is `retry-after-ms` or `retry-after` (both lower-cases). Not sure if this check will disregard case.